lib/srv/ctx.go

```diff
@@ -1330,3 +1330,36 @@ func (c *ServerContext) ConsumeApprovedFileTransferRequest() *FileTransferReques
 
 	return req
 }
+
+// The child does not signal until completing PAM setup, which can take an arbitrary
+// amount of time, so we use a reasonably long timeout to avoid dubious lockouts.
+const childReadyWaitTimeout = 3 * time.Minute
+
+// WaitForChild waits for the child process to signal ready through the named pipe.
+func (c *ServerContext) WaitForChild(ctx context.Context) error {
+	bpfService := c.srv.GetBPF()
+	pam, err := c.srv.GetPAM()
```

Comment:
It feels wrong to bail with an here with an error if PAM is not supported. What do you think about changing the GetPAM signature to remove the error and have any implementations that don't support PAM always return `servicecfg.PAMConfig.Enabled: false` instead of an error?